frontend-react/src/pages/Validate.tsx

```diff
@@ -0,0 +1,392 @@
+import React, { useState } from "react";
+import {
+    Button,
+    Form,
+    FormGroup,
+    Label,
+    FileInput,
+} from "@trussworks/react-uswds";
+import { useResource } from "rest-hooks";
+import moment from "moment";
+import { NavLink } from "react-router-dom";
+
+import SenderOrganizationResource from "../resources/SenderOrganizationResource";
+import { showError } from "../components/AlertNotifications";
+import Spinner from "../components/Spinner";
+import { watersApiFunctions } from "../network/api/WatersApiFunctions";
+import { useSessionContext } from "../contexts/SessionContext";
+
+// values taken from Report.kt
+const PAYLOAD_MAX_BYTES = 50 * 1000 * 1000; // no idea why this isn't in "k" (* 1024).
+const REPORT_MAX_ITEMS = 10000;
+const REPORT_MAX_ITEM_COLUMNS = 2000;
+// const REPORT_MAX_ERRORS = 100;
+
+const Validate = () => {
+    const [isSubmitting, setIsSubmitting] = useState(false);
```

Comment:
with this many separate state declarations, I'd recommend using a reducer via `useReducer`

Reply:
TODO in the next ticket :laughing:

Reply:
This is a direct carry-over from upload. Made a note in #5506 to refer back to comments here.